### ISSUE TITLE
precompile for armv6?

### ISSUE THREAD

**Issue by petermm** [cocoa-xu/tflite_beam#66]
any reason they are not precompiled presently? 

tflite_elixir seems to have armv6 support..

**Comment by cocoa-xu** [comment]
Ah yes, that one is specific to RPi (using `armv6-nerves-linux-gnueabihf-gcc` with its own optimisation args). I'm hesitating if it's worth to invest in this right now, given that most armv6 devices are relatively old and not that powerful. But for sure, this can be one of the todos for this project.

**Comment by cocoa-xu** [comment]
The other problem is using `armv6-nerves-linux-gnueabihf-gcc` for rpi with flags specified in [nerves_system_rpi](https://github.com/nerves-project/nerves_system_rpi/blob/243ca6624bfb8074a4754e7a16a655d6a06097de/mix.exs#L61C11-L61C140), it will generate binaries for `armv6kz`, which is the one for RPi.

```
$ readelf -A libedgetpu.so.1.0
Attribute Section: aeabi
File Attributes
  Tag_CPU_name: "6KZ"
  Tag_CPU_arch: v6KZ
  Tag_ARM_ISA_use: Yes
  Tag_THUMB_ISA_use: Thumb-1
  Tag_FP_arch: VFPv2
  Tag_ABI_PCS_wchar_t: 4
  Tag_ABI_FP_denormal: Needed
  Tag_ABI_FP_exceptions: Needed
  Tag_ABI_FP_number_model: IEEE 754
  Tag_ABI_align_needed: 8-byte
  Tag_ABI_align_preserved: 8-byte, except leaf SP
  Tag_ABI_enum_size: int
  Tag_ABI_VFP_args: VFP registers
  Tag_CPU_unaligned_access: v6
  Tag_ABI_FP_16bit_format: IEEE 754
  Tag_Virtualization_use: TrustZone
```

However, when using `arm-linux-gnueabihf-gcc` from the default apt repos 

```
apt install gcc-arm-linux-gnueabihf
```

with the same flags from [nerves_system_rpi](https://github.com/nerves-project/nerves_system_rpi/blob/243ca6624bfb8074a4754e7a16a655d6a06097de/mix.exs#L61C11-L61C140)

```
-mabi=aapcs-linux -mfpu=vfp -marm -fstack-protector-strong -mfloat-abi=hard -mcpu=arm1176jzf-s -fPIE -pie -Wl,-z,now -Wl,-z,relro
```

the binary it outputs is compiled for armv7a.

```
$ /usr/bin/arm-linux-gnueabihf-gcc -mabi=aapcs-linux -mfpu=vfp -marm -fstack-protector-strong -mfloat-abi=hard -mcpu=arm1176jzf-s -fPIE -pie -Wl,-z,now -Wl,-z,relro 1.c
$ readelf -A a.out
Attribute Section: aeabi
File Attributes
  Tag_CPU_name: "7-A"
  Tag_CPU_arch: v7
  Tag_CPU_arch_profile: Application
  Tag_ARM_ISA_use: Yes
  Tag_THUMB_ISA_use: Thumb-2
  Tag_FP_arch: VFPv3-D16
  Tag_ABI_PCS_wchar_t: 4
  Tag_ABI_FP_rounding: Needed
  Tag_ABI_FP_denormal: Needed
  Tag_ABI_FP_exceptions: Needed
  Tag_ABI_FP_number_model: IEEE 754
  Tag_ABI_align_needed: 8-byte
  Tag_ABI_align_preserved: 8-byte, except leaf SP
  Tag_ABI_enum_size: int
  Tag_ABI_VFP_args: VFP registers
  Tag_CPU_unaligned_access: v6
  Tag_Virtualization_use: TrustZone
```

Although I can just download the nerves toolchain, but it is essentially specific for the `arm1176jzf` CPU, i.e., Raspberry Pi (or perhaps we can even further relax it to any `armv6kz` CPU), but it's still not for any armv6 CPU.

I'm not an expert in compiler and compiler options, so if anyone has any idea/suggestion/comments on this, please feel free to reply.

**Comment by cocoa-xu** [comment]
Hi @petermm, precompiled version for armv6 is available now as in v0.3.5. I'm going to close this issue, but please free feel to reopen it if it doesn't work 😃
